src/System.Private.ServiceModel/tools/test/SelfHostWcfService/TestResources/EndpointResource.cs

```diff
@@ -110,5 +111,17 @@ private Uri BuildUri(ResourceRequestContext context)
             builder.Scheme = Protocol;
             return builder.Uri;
         }
+
+        private static string Getfqdn()
+        {
```

Comment:
The domain for cloud VMs is extremely unlikely to change as it will break a lot of existing customers. In my opinion, it only make senses to pass around if it's likely to change.